app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/RulesDialog.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ChoiceRule,
+  choiceRulesToConditonalRules,
+  cleanChoiceIdsFromRules,
+} from "@lib/formContext";
+import { getPathString } from "@lib/utils/form-builder/getPath";
+import { useRefsContext } from "@formBuilder/[id]/edit/components/RefsContext";
+
+type RulesDialogEventDetails = {
+  mode: "add" | "edit";
+  itemId: number;
+  optionId: string;
+};
+
+export const RulesDialog = () => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const { t, i18n } = useTranslation("form-builder");
+  const { Event } = useCustomEvent();
+  const dialog = useDialogRef();
+  const { togglePanel } = useTreeRef();
+  const router = useRouter();
+  const { refs } = useRefsContext();
+
+  const [mode, setMode] = React.useState<"add" | "edit">("add");
+  const [item, setItem] = React.useState<FormElementWithIndex | undefined>(undefined);
+  const [optionId, setOptionId] = React.useState<string | null>(null);
+
+  const choiceRulesRef = React.useRef<ChoiceRule[]>([]);
+
+  const { getFormElementWithIndexById, elements, formId, updateField } = useTemplateStore((s) => ({
+    getFormElementWithIndexById: s.getFormElementWithIndexById,
+    elements: s.form.elements,
+    formId: s.id,
+    updateField: s.updateField,
+  }));
+
+  // @TODO: check?
+  // if (storeId && storeId !== formId) {
+  //   formId = storeId;
+  // }
+
+  // const isRichText = item?.type == "richText"; // @TODO: check?
+  const descriptionId = `descriptionId-${Date.now()}`;
+
+  const handleOpenDialog = useCallback(
+    (detail: RulesDialogEventDetails) => {
+      if (detail) {
+        const freshItem = getFormElementWithIndexById(detail.itemId);
+        setItem(freshItem);
+        setMode(detail.mode);
+        setOptionId(detail.optionId);
+        setIsOpen(true);
+      }
+    },
+    [getFormElementWithIndexById]
+  );
+
+  const handleSubmit = ({ item }: { item: FormElementWithIndex }) => {
+    // Handle the case where there are no rules
+    if (!choiceRulesRef.current || !choiceRulesRef.current.length) {
+      elements.forEach((el) => {
+        if (!el.properties.conditionalRules) return;
+        const rules = cleanChoiceIdsFromRules(String(item.id), el.properties.conditionalRules);
+
+        // Just return if updated rules match existing rules
+        if (JSON.stringify(rules) === JSON.stringify(el.properties.conditionalRules)) return;
+
+        // Update the element rules if they have changed
+        const properties = { ...el.properties, conditionalRules: rules };
+        updateField(getPathString(el.id, elements), properties);
+      });
+      setIsOpen(false);
+      return;
+    }
+
+    const els = choiceRulesToConditonalRules(elements, choiceRulesRef.current);
+    // Update the element rules
+    Object.entries(els).forEach(([key, value]) => {
+      const element = elements.find((el) => el.id == Number(key));
+      if (!element) return;
+      const properties = { ...element.properties, conditionalRules: value };
```

Comment:
value should probably be  rules here

Reply:
f142e551781be5a10e73d6fbf43586a600360f73